### PR TITLE
ticdc: add index design for cdc storage sink

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Added description about index.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/13448
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by CharlesCheung96 — APPROVED** [review]

**Review by Frank945946 — APPROVED** [review]

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 1]
```diff
@@ -101,6 +97,22 @@ Data change records are saved to the following path:
...
+{scheme}://{prefix}/{schema}/{table}/{table-version-separator}/{partition-separator}/{date-separator}/CDC.index
+```
+
+An index file records the largest file name used in the current directory. For example:
+
+```
+CDC000005.csv
+```
+
+In this example, the files CDC000001.csv through CDC000004.csv in this directory are occupied. When a table scheduling or node restart occurs in the TiCDC cluster, the new node reads the index file and determines if CDC000005.csv is occupied. If it is not occupied, the new node writes the file starting from CDC000005.csv. If it is occupied, it starts writing from CDC000006.csv, which prevents overwriting data written by other nodes.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:114]:
```suggestion
In this example, the files `CDC000001.csv` through `CDC000004.csv` in this directory are occupied. When a table scheduling or node restart occurs in the TiCDC cluster, the new node reads the index file and determines if `CDC000005.csv` is occupied. If it is not occupied, the new node writes the file starting from `CDC000005.csv`. If it is occupied, it starts writing from `CDC000006.csv`, which prevents overwriting data written by other nodes.
```

File: ticdc/ticdc-csv.md [hunk 2]
```diff
@@ -35,6 +35,10 @@ include-commit-ts = true
 - The same transactions of a single table are stored in the same CSV file.
 - Multiple tables of the same transaction can be stored in different CSV files.
 
+## Data storage path structure
+
+For more information about the storage path structure of data change records, metadata, and DDL events, see [Storage path structure](/ticdc/ticdc-sink-to-cloud-storage.md#storage-path-structure).
```
Comment (hfxsd) [inline ticdc/ticdc-csv.md:40]:
```suggestion
For more information about the storage path structure of the data, see [Storage path structure](/ticdc/ticdc-sink-to-cloud-storage.md#storage-path-structure).
```

[the remark this answers is not in the document]
Comment (hfxsd) [reply ticdc/ticdc-csv.md:40]: Updated

**Review by qiancai — APPROVED** [review]
LGTM

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 3]
```diff
@@ -5,11 +5,7 @@ summary: Learn how to replicate data to storage services using TiCDC, and learn
 
 # Replicate Data to Storage Services
 
-> **Warning:**
->
-> This feature is experimental. It is not recommended to use it in the production environment.
-
-Since v6.5.0, TiCDC supports saving row change events to storage services, including Amazon S3, Azure Blob Storage, and NFS. This document describes how to create a changefeed that replicates incremental data to such storage services using TiCDC, and how data is stored. The organization of this document is as follows:
+Since v6.5.0, TiCDC supports saving row change events to storage services, including Amazon S3, GCS, Azure Blob Storage, and NFS. This document describes how to create a changefeed that replicates incremental data to such storage services using TiCDC, and how data is stored. The organization of this document is as follows:
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-cloud-storage.md:8]:
```suggestion
Starting from TiDB v6.5.0, TiCDC supports saving row change events to storage services, including Amazon S3, GCS, Azure Blob Storage, and NFS. This document describes how to create a changefeed that replicates incremental data to such storage services using TiCDC, and how data is stored. The organization of this document is as follows:
```

**Pull request merged**